src/Language/Fixpoint/SortCheck.hs

```diff
@@ -960,9 +961,26 @@ throwErrorAt ~err' = do -- Lazy pattern needed because we use LANGUAGE Strict in
 -- | Helper for checking symbol occurrences
 checkSym :: Env -> Symbol -> CheckM Sort
 checkSym f x = case f x of
-  Found s -> instantiate s
+  Found s -> refreshNegativeTyVars s >>= instantiate
   Alts xs -> throwErrorAt (errUnboundAlts x xs)
 
+-- Negative type variables are implictly universally quantified type variables
```

Comment:
Insight like this should be explained in the documentation of `Sort` data type, so people know how to use it to represent sorts.

The code itself looks good.